[Expressions] Add a flag to opt-out of the partial results in the expressions

dokmic opened the pull request:
## Summary

Resolves #143344.

### Checklist

- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServicesSv)

Dosant commented:
> Resolves https://github.com/elastic/kibana/pull/143444.

Is this is the right issue? Doesn't look like it

Dosant commented:
Trying to understand: 
Previously we had `partial` switch inside the expression loader. Looks like it was false by default and that's why we didn't show partial results in charts. 

Now we also add the switch to the execution: why? what is the benefit? How this flag related to the one in the expression loader? Why it is by default true for the execution, but false for the loader ?

dokmic commented:
@Dosant Thanks for this finding. That makes total sense.

I've reused the newly added flag in the loader and changed the default value to `false`. Considering #141673 and #143344, it makes sense to disable partial results to avoid potential performance issues and side effects since we are not using them yet. My only concern is that we had it turned on for a while, and now it's changed. WDYT?

Additionally, I've moved the `throttle` parameter to the execution context to align with the `partial` flag.

flash1293 commented:
@Dosant could you take another look at this? We depend on it for the metric trendlines

Dosant commented:
> My only concern is that we had it turned on for a while, and now it's changed. WDYT?

In my opinion, If one actually uses them, there is no harm in disabling them on the expression layer by default as this adds overhead. 
And if someone needs it, then can enable it by using the flag, right? 

I'll run and play with it short

flash1293 commented:
+1 for disabling by default.

Dosant reviewed (approved):
LGTM 👍

andrewctate commented on src/plugins/expressions/common/execution/execution.test.ts:
```diff
@@ -426,6 +426,68 @@ describe('Execution', () => {
       });
     });
+
+    test('ignores partial results by default', () => {
```
So, I guess this means that we won't have to do anything on the Lens side to take advantage of this change.
